smoke-test/src/test/java/org/opennms/smoketest/TopologyIT.java

```diff
@@ -0,0 +1,383 @@
+            }
+            return visibleVertices;
+        }
+
+        public int getSzl() {
+            return Integer.parseInt(testCase.findElementById("szlInputLabel").getText());
+        }
+
+        public TopologyUiPage setSzl(int szl) {
+            testCase.enterText(By.id("szlInputLabel"), Integer.toString(szl));
+            waitForTransition();
+            return this;
+        }
+
+        /**
+         * This method is used to block and wait for any transitions to occur.
+         * This should be used after adding or removing vertices from focus and/or
+         * changing the SZL.
+         */
+        private void waitForTransition() {
+            try {
+                // TODO: Find a better way that does not require an explicit sleep
+                Thread.sleep(2000);
+            } catch (InterruptedException e) {
+                throw Throwables.propagate(e);
+            }
+        }
+
+        private WebElement getMenubarElement(String itemName) {
+            return testCase.findElementByXpath("//span[@class='v-menubar-menuitem-caption' and text()='" + itemName + "']/parent::*");
+        }
+
+        private WebElement getShowEntireMapElement() {
+            return testCase.findElementByXpath("//i[@class='icon-globe']");
+        }
+
+        private void resetMenu() {
+            // The menu can act weirdly if we're already hovering over it so we mouse-over to
+            // a known element off of the menu, and click a couples times just to make sure
+            WebElement showEntireMap = getShowEntireMapElement();
+            Actions actions = new Actions(testCase.m_driver);
+            actions.moveToElement(showEntireMap);
+            actions.clickAndHold();
+            try {
+                Thread.sleep(500);
+            } catch (InterruptedException e) {
+                throw Throwables.propagate(e);
+            }
+            actions.release();
+            showEntireMap.click();
+        }
+
+        private boolean isMenuItemChecked(String itemName, String... path) {
+            clickOnMenuItemsWithLabels(path);
+
+            final WebElement automaticRefresh = getMenubarElement(itemName);
+            final String cssClasses = automaticRefresh.getAttribute("class");
+            if (cssClasses != null) {
+                if (cssClasses.contains("-unchecked")) {
+                    return false;
+                } else if (cssClasses.contains("-checked")) {
+                    return true;
+                } else {
+                    throw new RuntimeException("Unknown CSS classes '" + cssClasses + "'."
+                                + " Unable to determine if the item is checked or unchecked.");
+                }
+            } else {
+                throw new RuntimeException("Element has no CSS classes!"
+                        + " Unable to determine if the item is checked or unchecked.");
+            }
+        }
+    }
+
+    public static class TopologyUiSearchResults {
```

Comment:
Can we rename this to `TopologyUISearchResults` just to be consistend with the `TopologyUI` class?